File: src/main/java/p00/Problem_0033.java
package p00;

public class Problem_0033 {

    public static int solve() {

        int numerator = 1;
        int denominator = 1;

        for (int i = 10; i < 99; i++)
            for (int j = i + 1; j < 100; j++)

                if ((("" + i).charAt(1) == ("" + j).charAt(0) && ("" + i).charAt(1) != '0') && Double.parseDouble("" + ("" + i).charAt(0)) / Double.parseDouble("" + (("" + j)).charAt(1)) == i
                        / (double) j) {

                    numerator *= i;
                    denominator *= j;
                }

        return denominator / gcd(numerator, denominator);
    }

    private static int gcd(int pA, int pB) {
        return pA == pB ? pA : pA > pB ? gcd(pA - pB, pB) : gcd(pB - pA, pA);
    }

}

File: src/main/kotlin/p00/Problem0033.kt
package p00

class Problem0033 {
    fun solve(): Int {
        var numerator = 1
        var denominator = 1
        for (i in 10..98) for (j in i + 1..99) if (("" + i)[1] == ("" + j)[0] && ("" + i)[1] != '0' && ("" + ("" + i)[0]).toDouble() / ("" + ("" + j)[1]).toDouble() == i
                / j.toDouble()) {
            numerator *= i
            denominator *= j
        }
        return denominator / gcd(numerator, denominator)
    }

    private fun gcd(pA: Int, pB: Int): Int {
        return if (pA == pB) pA else if (pA > pB) gcd(pA - pB, pB) else gcd(pB - pA, pA)
    }
}